Add guest records to /etc/hosts

jhriv opened the pull request:
Allow guests to refer to each other by hostname

Bump minor version

Support DHCP assigned private networking

Updated documentation

Reduce apt verbosity

  Work around a dpkg quirk, when forked with a pty, will display database
  progress regardless of requested --quiet level
  https://askubuntu.com/questions/258219/how-do-i-make-apt-get-install-less-noisy

  Silences a stdin warning
  https://serverfault.com/questions/500764/dpkg-reconfigure-unable-to-re-open-stdin-no-file-or-directory

jhriv committed:
Add guest records to /etc/hosts

Allow guests to refer to each other by hostname

Bump minor version

Support DHCP assigned private networking

Updated documentation

Reduce apt verbosity

  Work around a dpkg quirk, when forked with a pty, will display database
  progress regardless of requested --quiet level
  https://askubuntu.com/questions/258219/how-do-i-make-apt-get-install-less-noisy

  Silences a stdin warning
  https://serverfault.com/questions/500764/dpkg-reconfigure-unable-to-re-open-stdin-no-file-or-directory

mcritchlow commented on Makefile:
```diff
@@ -90,6 +90,9 @@ ip: ansible.cfg
 python: ansible.cfg
 	@ansible all -m raw -a 'sudo apt-get install --assume-yes python python-apt'
 
+etc-hosts: ansible.cfg
```
Should this be part of `all`?

jhriv replied:
I waffled on that.

I also waffled on naming it `.etc-hosts.yml` or otherwise hiding the playbook to keep the directory as clean as possible.

In the end - all the Makefile itself needs is itself, and a valid Vagrantfile (think `vagrant init`). An internet connection is not even required. Making etc-hosts part of all requires another file, or the ability (not yet implemented) to download it (à la `Vagrantfile-force`).

Either way, I do need to update the `make menu` target, add the `etc-hosts.yml` role to download, use tags to download (instead of hard coding master), and tag the commit.

mcritchlow commented on etc-hosts.yml:
```diff
@@ -0,0 +1,40 @@
+---
+
+- name: Update /etc/hosts in guests
```
I think this is really clever.. The only thing I can think of that would be a _possible_ alternative would be to do a custom fact script that would manage the logic. That said.. you'd still need to use those facts to populate this file, so I suspect this is probably still the optimal solution.

jhriv committed:
Add etc-hosts to menu

Renamed automatically created files
Refactored to use variables instead of hardcoding filenames.
Downloads (except for update) use tag (which should match $VERSION)
Alphabetized global variable assignments
Moved force-role taget out of “all: list in menu target; it was never part of the all target
Corrected SSHCONF->SSHCONFIG in example make invocation
